src/main/java/uk/gov/hmcts/darts/audio/component/impl/AudioMessageDigestImpl.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.darts.audio.component.impl;
+
+import uk.gov.hmcts.darts.audio.component.AudioMessageDigest;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+
+import java.security.NoSuchAlgorithmException;
+
+import static uk.gov.hmcts.darts.audio.exception.AudioApiError.FAILED_TO_UPLOAD_AUDIO_FILE;
+
+public class AudioMessageDigestImpl implements AudioMessageDigest {
+
+    private final String algorithm;
+
+    public AudioMessageDigestImpl(String algorithm) {
+        this.algorithm = algorithm;
+    }
```

Comment:
Consider replacing the explicit constructor with a `@RequiredArgsConstructor` annotation at class level for slightly less verbosity.